react/utils/autoComplete.tsx

```diff
@@ -0,0 +1,79 @@
+import type { FC } from 'react'
+import React, { useRef, useState, useContext } from 'react'
+import { AutocompleteInput } from 'vtex.styleguide'
+
+import Context from '../Context/context'
+
+interface Props {
+  onChange: any
+  name: string
+}
+
+const AutoComplete: FC<Props> = ({
+  onChange,
+  name,
+}: {
+  onChange: any
+  name: string
+}) => {
+  const [term, setTerm] = useState('')
+  const [loading, setLoading] = useState(false)
+  const timeoutRef = useRef<any>()
+
+  const provider = useContext(Context)
+  let values
+
+  if (name === 'productId') values = provider.nameProducts
+  if (name === 'selectedItemId') values = provider.nameSku
+  if (name === 'brandId') values = provider.nameBrands
+  if (name === 'productClusters') values = provider.nameCollections
+  if (name === 'categoryId') values = provider.nameCategory
+  if (name === 'specificationProperties') values = provider.nameSpecification
```

Comment:
It's a better idea use dictionary in this case, to avoid multiples `if` checks

```suggestion
  enum fieldNames {
    productId,
    selectedItemId,
    brandId,
    productClusters,
    categoryId,
    specificationProperties,
  }

 ...

  const fields: { [key in keyof typeof fieldNames]: string } = useMemo(() => {
    return {
      productId: provider.nameProducts,
      selectedItemId: provider.nameSku,
      brandId: provider.nameBrands,
      productClusters: provider.nameCollections,
      categoryId: provider.nameCategory,
      specificationProperties: provider.nameSpecification,
    }
  }, [
    provider.nameBrands,
    provider.nameCategory,
    provider.nameCollections,
    provider.nameProducts,
    provider.nameSku,
    provider.nameSpecification,
  ])

  const values = useMemo(() => fields[name], [fields, name])
```